packages/accounts-password/password_server.js

```diff
@@ -500,11 +500,16 @@ Accounts.setPassword = function (userId, newPlaintextPassword, options) {
 Meteor.methods({forgotPassword: function (options) {
   check(options, {email: String});
 
-  var user = Meteor.users.findOne({"emails.address": options.email});
+  var user = Accounts._findUserByQuery({email: options.email});
   if (!user)
     throw new Meteor.Error(403, "User not found");
 
-  Accounts.sendResetPasswordEmail(user._id, options.email);
+  const emails = _.pluck(user.emails || [], 'address');
+  const caseSensitiveEmail = _.find(emails, email => {
```

Comment:
The reason I put this logic inside of `sendResetPasswordEmail` was so that when a developer from server-side code manually are using `sendResetPasswordEmail` doesn't have to care about finding and using the caseSensisiveEmail.

Which they will have to do now, otherwise the `resetPassword` method will fail, as it expects the email in the resetToken to be of the same case.